src/sentry/uptime/endpoints/validators.py

```diff
@@ -497,5 +529,11 @@ def update_data_source(self, instance: Detector, data_source: dict[str, Any]):
 
         return instance
 
-    def create(self, validated_data):
-        return super().create(validated_data)
+    def delete(self) -> None:
+        assert self.instance is not None
+
+        remove_uptime_seat(self.instance)
+        uptime_subscription = get_uptime_subscription(self.instance)
+        delete_uptime_subscription(uptime_subscription)
+
+        super().delete()
```

Comment:
👀 yeah, this does feel a bit cleaner than having the hooks -- yay it's just one surface area now! 🎉